packages/wdio-utils/src/test-framework/testFnWrapper.ts

```diff
@@ -109,6 +109,8 @@ export const testFrameworkFnWrapper = async function (
     if (error && !error.matcherName) {
         throw error
     }
+    //Handle jest-snapshot suppressed errors
+    if (this.snapshotError) { throw this.snapshotError }
```

Comment:
This should be handled by the service. It should use the `afterTest` or `afterHook` hook to verify that no errors were thrown. The service should throw a `SevereServiceError` to make the test fail.

Reply:
Tried throw SevereServiceError in the aftertTest but seem the report couldn't pick up fail result
<img width="782" alt="image" src="https://github.com/webdriverio/webdriverio/assets/19924396/26c06318-5e66-4702-af9c-7662d320cfe3">
<img width="560" alt="image" src="https://github.com/webdriverio/webdriverio/assets/19924396/5496ac28-3e6c-44e4-a8f4-e3bead2918a0">